.config/semgrep.yaml

```diff
@@ -8,6 +8,21 @@ rules:
 
   severity: WARNING
 
+- id: ban-system-time-elapsed
+  languages:
+    - rust
+  message: Use fedimint_core::time::now and std::time::duration_since to compare an elapsed time for better wasm compatibility.
+  pattern: $SYSTEM_TIME.elapsed()
```

Comment:
I was hoping:

```
- id: ban-system-time-elapsed
  languages:
    - rust
  message: TBD
  pattern-either:
    - pattern: std::time::SystemTime::elapsed
    - pattern: std::time::Instant::elapsed

  severity: WARNING
```

would work, but it seems it doesn't.

Reply:
Yeah, I was hoping the same and came to the same conclusion for method calls via `$foo.bar`.

It's also worth noting a footgun:

```
pattern: '($SYSTEM_TIME: std::time::SystemTime).elapsed()'
```

This is valid syntax for typed metavariables and implies it will only match method calls to `SystemTime.elapsed()`, however semgrep is unable to parse the type from struct fields so we could accidentally sneak in additional calls to `.elapsed()`.